src/crc32.c

```diff
@@ -193,20 +193,28 @@ bool bmd_crc32(target_s *const target, uint32_t *const result, const uint32_t ba
 #if !defined(STM32F0) && !defined(STM32F1) && !defined(STM32F2) && !defined(STM32F3) && !defined(STM32F4) && \
 	!defined(STM32F7) && !defined(STM32L0) && !defined(STM32L1) && !defined(STM32G0) && !defined(STM32G4)
 	const bool status = generic_crc32(target, result, base, len);
+	const char *const func = "generic_crc32";
 #else
 	const bool status = stm32_crc32(target, result, base, len);
+	const char *const func = "stm32_crc32";
 #endif
 #ifndef DEBUG_INFO_IS_NOOP
-	/* "generic_crc32: 08000110+75272 -> 1353ms, 54 KiB/s" */
-	/* "stm32_crc32: 08000110+75272 -> 237ms, 310 KiB/s" */
+	/* "generic_crc32: 0x08000110+75272 -> 1353ms, 54 KiB/s" */
+	/* "stm32_crc32: 0x08000110+75272 -> 237ms, 310 KiB/s" */
 	const uint32_t end_time = platform_time_ms();
 	const uint32_t time_elapsed = end_time - start_time;
-	DEBUG_INFO("%s: 0x%08" PRIx32 "+%" PRIu32 " -> %" PRIu32 "ms", __func__, base, (uint32_t)len, time_elapsed);
-	if (len >= 512U) {
+	char logbuf[64] = {0};
+	const size_t logcap = 64;
+	size_t loglen = 0;
+	loglen = snprintf(&logbuf[loglen], logcap, "%s: 0x%08" PRIx32 "+%" PRIu32 " -> %" PRIu32 "ms", func, base,
```

Comment:
because of how this is used, we would strongly suggest expressing the first parameter as `logbuf + loglen` please - it won't affect the operation performed but is a little less brain-melting to follow.

Do note that the result of `snprintf()` is `int`/`ssize_t`, so the result *must* be cast to `size_t` to prevent a signed-unsigned conversion warning. The conversion is fine here, this is just to fix the lint this introduces.

Reply:
`snprintf` returns int, so maybe use `ssize_t` in the first place for loglen, logcap?